core/src/main/java/org/elasticsearch/search/aggregations/support/InternalOrder.java

```diff
@@ -327,40 +342,82 @@ public static void writeOrder(Terms.Order order, StreamOutput out) throws IOExce
                 out.writeString(path.toString());
             } else if (order instanceof CompoundOrder) {
                 CompoundOrder compoundOrder = (CompoundOrder) order;
-                    out.writeByte(order.id());
-                    out.writeVInt(compoundOrder.orderElements.size());
-                    for (Terms.Order innerOrder : compoundOrder.orderElements) {
-                        Streams.writeOrder(innerOrder, out);
-                    }
+                out.writeByte(order.id());
+                out.writeVInt(compoundOrder.orderElements.size());
+                for (BucketOrder innerOrder : compoundOrder.orderElements) {
+                    Streams.writeOrder(innerOrder, out);
+                }
             } else {
                 out.writeByte(order.id());
             }
         }
 
-        public static Terms.Order readOrder(StreamInput in) throws IOException {
-            return readOrder(in, false);
-        }
-
-        public static Terms.Order readOrder(StreamInput in, boolean absoluteOrder) throws IOException {
+        public static BucketOrder readOrder(StreamInput in) throws IOException {
+            // TODO backwards compat
```

Comment:
You'll need to add a check for the version from the stream, so something like:
```
if (in.getVersion().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED)) {
    <NEW SERIALISATION CODE GOES HERE>
} else {
    <BACKWARDS COMPATIBLE SERIALISATION CODE GOES HERE>
}
```

Have a look at the `IncludeExclude` class for and example. You will obviously need to do a similar thing for the `writeTo` method